readthedocs/core/resolver.py

```diff
@@ -214,6 +214,58 @@ def resolve(
         )
         return urlunparse((protocol, domain, path, '', query_params, ''))
 
+    def _get_root_path(self, project):
+        """
+        Returns the root path for a project.
+
+        If the project is a subproject, it will return ``/projects/<project-slug>/``.
+        If the project is a main project, it will return ``/``.
+        This will respect the custom urlconf of the project if it's defined.
+        """
+        custom_prefix = None
+        if project.urlconf:
+            custom_prefix = project.urlconf.split("$", 1)[0]
+
+        if project.is_subproject:
+            prefix = custom_prefix or "projects"
+            return join_url_path(prefix, project.slug, "/")
+
+        prefix = custom_prefix or "/"
+        return join_url_path(prefix, "/")
+
+    def resolve_root(self, project, external_version_slug=None):
+        """
+        Get the root URL from where the documentation of ``project`` is served from.
+
+        This doesn't include the version or language. For example:
+
+        - https://docs.example.com/projects/<project-slug>/
+        - https://docs.readthedocs.io/
+
+        This will respect the custom urlconf of the project if it's defined.
+
+        :param project: Project object to get the root URL from
+        :param external_version_slug: If given, resolve using the external version domain.
+        """
+        canonical_project = self._get_canonical_project(project)
+        use_custom_domain = self._use_cname(canonical_project)
+        custom_domain = canonical_project.get_canonical_custom_domain()
```

Comment:
Seems odd to have both of these... should `custom_domain` not take this into account? Should we pass use_custom_domain into`.get_canonical_custom_domain()` 

Who is responsible for deciding if a domain is used?

Reply:
We override the `_use_cname` method on .com to check for the plan.

Reply:
What about overriding `.get_canonical_custom_domain()` instead? Would that be doable and simpler?

Reply:
Once we unify feature check, we won't be needing overrides.